### PR TITLE
Finally Servo works with newer winit

### PR DESCRIPTION
**Pull request by Snowiiii**
Update Dependencies & Rust Edition 

Using Surfman cargo override for winit update 
(yay servo on linux)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #[29149](https://github.com/servo/servo/issues/29149) 

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because Dependency update

### CODE REVIEW

**Comment by jdm** [comment]
Could you revert all the rust edition changes and updates to dependencies that are not required to upgrade winit? By making the changes as small as possible, it makes it easier to identify the cause of any regressions.

**Comment by Snowiiii** [comment]
i put much time to Update the Dependencies so that all is working fine and Rust Edition 2021 is just the "Standart" ,CL is unhappy because of an CMake issue if i see correct
are you sure what i should only use it for winit ? if you want you can test the build yourself

**Comment by jdm** [comment]
This is a change that could affects every platform differently. Theore changes the make at the same time, the harder it is to know which change is actually responsible for a problem. The best practice is to start by updating only winit, and see if anything else needs to change.

**Comment by Snowiiii** [comment]
Done and works

**Comment by jdm** [comment]
https://github.com/Snowiiii/servo/pull/1 has fixes for macOS.

**Comment by Snowiiii** [comment]
Merged

**Comment by okias** [comment]
Nice.

**Comment by Snowiiii** [comment]
If i understand the Check fails becausd of CMake issue ?

**Comment by jdm** [comment]
Huh, that's new:
```
   CMake Error at cmake_install.cmake:46 (file):
    file INSTALL cannot copy file
    "/home/runner/work/servo/servo/target/release/build/fontsan-28a8f43cbd285306/out/build/libots.a"
    to "/usr/local/lib/libots.a": Success.
```
We should check whether it's reprodicible in the CI for https://github.com/servo/fontsan.

**Comment by jdm** [comment]
Huh. Seems like it's something about the changes in this PR in particular, since #29184 just merged successfully.

File: Cargo.toml [hunk 1]
```diff
@@ -16,20 +16,7 @@ opt-level = 3
...
-#
-#     <crate> = { path = "/path/to/local/checkout" }
-#
-# Or for a git dependency:
-#
-#     [patch."https://github.com/servo/<repository>"]
-#     <crate> = { path = "/path/to/local/checkout" }
-
-# This is here to dedupe winapi since mio 0.6 is still using winapi 0.2.
-mio = { git = "https://github.com/servo/mio.git", branch = "servo-mio-0.6.22" }
```
Comment (atouchet) [inline Cargo.toml:30]: Is this mio fork supposed to be removed here? Isn't it still required for using a compatible version of winapi?

Comment (jdm) [reply Cargo.toml:30]: The newer versions of winit use a newer version of mio, so the fork can no longer be used. It was only necessary for UWP builds, and we're not shipping those in the Windows app store any more, so it's less important.

Comment (atouchet) [reply Cargo.toml:30]: mio 0.6 is still being used as a dependency of other crates though.

**Comment by Snowiiii** [comment]
was gone a bit and was not able to code because im sick,The new fontsan patch passed the check fine

**Comment by Snowiiii** [comment]
so what now.CI does not make sense if you want i can show you that its building fine locally

**Comment by jdm** [comment]
```
[[patch.unused]]
name = "fontsan"
version = "0.4.0"
source = "git+https://github.com/Snowiiii/fontsan.git?branch=master#2ef5bc9ba3a58912189f8c8b1b18361a79bac40f"
```
This means the override isn't actually being used for some reason.

**Comment by Snowiiii** [comment]
so @jdm what to do next i dont know what the patch is unused

**Comment by atomgardner** [comment]
@Snowiiii, are you still working on this?

**Comment by Snowiiii** [comment]
Hey sorry but i guess im done with Servo,Servo is very Compilcated and currently i dont have desire or Time to work on it, But it would be Very Nice if someone else Port to Winit 0.28

**Pull request closed without merging**